plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/TemplateViewerFilter.java

```diff
@@ -38,15 +38,19 @@ public boolean select(Viewer viewer, Object parentElement, Object element) {
 		if(!(element instanceof IResource)) {
 			return false;
 		}
+		final String text = filterText.getText();
+		if(StringUtils.isBlank(text)) {
+			return true;
+		}
 		IResource resource = (IResource) element;
-		if(StringUtils.isBlank(filterText.getText())) {
+		if(resource.getName().contains(text)) {
```

Comment:
A resource needs a name for it to exist in the system so it can never be 
null.  Also, our client impl using DMR doesn't allow for it though this 
is not an excuse either.  Technically, we should probably invert the 
check since we already check the filterText for null.

On 06/09/2015 08:20 AM, André Dietisheim wrote:
> In
> plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/TemplateViewerFilter.java
> <https://github.com/jbosstools/jbosstools-openshift/pull/652#discussion_r32006550>:
>
>>  		IResource resource = (IResource) element;
>> -		if(StringUtils.isBlank(filterText.getText())) {
>> +		if(resource.getName().contains(text)) {
>
> can't #getName return null?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jbosstools/jbosstools-openshift/pull/652/files#r32006550>.
>

-- 
Jeff Cantrill
Senior Software Engineer, Red Hat Engineering
Red Hat
Office: 703-748-4420 | 866-546-8970 ext. 8162420
jcantril@redhat.com
http://www.redhat.com